src/OrbitGl/TimeGraph.cpp

```diff
@@ -969,12 +928,34 @@ std::pair<const TimerInfo*, const TimerInfo*> TimeGraph::GetMinMaxTimerInfoForFu
   return std::make_pair(min_timer, max_timer);
 }
 
-void TimeGraph::DrawText(float layer) {
-  if (draw_text_) {
-    text_renderer_static_.RenderLayer(layer);
+void TimeGraph::DoDraw(Batcher& batcher, TextRenderer& text_renderer,
+                       const DrawContext& draw_context) {
+  CaptureViewElement::DoDraw(batcher, text_renderer, draw_context);
+
+  DrawIncompleteDataIntervals(batcher, draw_context.picking_mode);
+  DrawOverlay(batcher, text_renderer, draw_context.picking_mode);
+}
+
+void TimeGraph::DrawAllElements(Batcher& batcher, TextRenderer& text_renderer,
```

Comment:
I'm not following 100% why we need this function and can't move this to the overloaded DoDraw instead.

Reply:
`TimeGraph` is currently the root of our whole element-tree in the capture window, and this is the only public API method used for drawing. All others are protected. `DrawElements` takes care of calling `Draw` and `UpdatePrimitives` when needed, so from outside of `TimeGraph´, there is no need to know about the difference between those two. Also the outside world does not need to know about the min / max_tick parameters of `UpdatePrimitives`, because this is something internal to `TimeGraph`.

This might be adjusted in the future, it could e.g. be that when we refactor the time bar, `TimeGraph` is no longer the root. Currently, `TimeGraph` mixes up too many functionalities, probably this should be moved out into a UI element that simply acts as a root of all elements and takes care of scrolling, nothing else.

Reply:
Really thanks for the explanation. Makes a lot of sense. :)